x/concentrated-liquidity/position_test.go

```diff
@@ -91,17 +105,54 @@ func (s *KeeperTestSuite) TestInitOrUpdatePosition() {
 			// Init suite for each test.
 			s.Setup()
 
+			// Set blocktime to fixed UTC value for consistency
+			s.Ctx = s.Ctx.WithBlockTime(time.Unix(1, 1).UTC())
+
 			// Create a default CL pool
-			s.PrepareConcentratedPool()
+			clPool := s.PrepareConcentratedPool()
+
+			// We get initial uptime accum values for comparison later
+			initUptimeAccumValues, err := s.App.ConcentratedLiquidityKeeper.GetUptimeAccumulatorValues(s.Ctx, test.param.poolId)
+			if test.param.poolId == invalidPoolId {
+				s.Require().Error(err)
+				// Ensure that no accumulators are retrieved upon error
+				s.Require().Equal([]sdk.DecCoins{}, initUptimeAccumValues)
+			} else {
+				s.Require().NoError(err)
+				// Ensure initial uptime accums are empty
+				s.Require().Equal(getExpectedUptimes().emptyExpectedAccumValues, initUptimeAccumValues)
+			}
+
+			// Set incentives for pool to ensure accumulators work correctly
+			s.App.ConcentratedLiquidityKeeper.SetMultipleIncentiveRecords(s.Ctx, test.incentiveRecords)
 
 			// If positionExists set, initialize the specified position with defaultLiquidityAmt
 			preexistingLiquidity := sdk.ZeroDec()
 			if test.positionExists {
+				// We let some fixed amount of time to elapse so we can ensure LastLiquidityUpdate time is
+				// tracked properly even with no liquidity.
+				s.Ctx = s.Ctx.WithBlockTime(s.Ctx.BlockTime().Add(time.Minute * 5))
+
 				err := s.App.ConcentratedLiquidityKeeper.InitOrUpdatePosition(s.Ctx, test.param.poolId, s.TestAccs[0], test.param.lowerTick, test.param.upperTick, test.param.liquidityDelta, test.param.frozenUntil)
 				s.Require().NoError(err)
 				preexistingLiquidity = DefaultLiquidityAmt
+
```

Comment:
We should probably be setting `preexistingLiquidity` to `test.param.liquidityDelta` instead of `DefaultLiquidityAmt` right? Even though they are currently equal, we might add a different test case in the future that would break this.

Reply:
Agreed, changed